admin/main.js

```diff
@@ -1,85 +1,76 @@
+    addUserWidget.hidden = !addUserWidget.hidden;
+  });
   const logoutButton = document.getElementById('logout')
-  const loginButton = document.getElementById('login');
+  const addUserButton = document.getElementById('add-user-button');
 
   function postData(url = '', data = {}) {
-      fetch(url, {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/json'
-          },
-          body: JSON.stringify(data)
-      }).then(response => {
-        response.json()
-        if (response.ok) {
-          location.reload(true);
-        };
-        console.log(response)
-        return response
-      } )     //  return response.json();
+    return fetch(url, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(data)
+    });
   }
-  fetch('/admin/rpc/list-users').then((response) => {
-    response.json().then(resp => {
-      for (const user of resp) {
-        const li = document.createElement('li');
-        const button = document.createElement('button');
-        const spanName = document.createElement('span');
-        const spanButton = document.createElement('span');
-        const strong = document.createElement('strong');
-        button.innerText ='x';
-        button.className = 'list'
-        spanButton.className = 'button'
-        spanName.innerText = user.name + ' ' + user.lastName;
-        li.appendChild(spanName);
-        if(username !== user.username) {
-          spanButton.appendChild(button);
-          li.appendChild(spanButton);
-          list.appendChild(li);
-          
-          button.addEventListener("click",  () => {
-            postData('/admin/rpc/delete-users', {
-                userID: user.userID
-            });
-          });
-        } else {
-          strong.innerText ='Admin';
-          spanButton.appendChild(strong);
-          li.appendChild(spanButton);
-          list.appendChild(li);
-          
-          button.addEventListener("click",  () => {
-            postData('/admin/rpc/delete-users', {
-                userID: user.userID
-            });
-          });
-        }
+
+  function refresh() {
+    list.innerHTML = '';
+    for (const user of users) {
+      const li = document.createElement('li');
+      const spanName = document.createElement('span');
+      const spanButton = document.createElement('span');
+      spanButton.className = 'button';
+      spanName.innerText = user.name + ' ' + user.lastName;
+      li.appendChild(spanName);
+      if (username !== user.username) {
+        spanButton.innerHTML = `<button class="list" data-id="${user.userID}">x</button>`;
+      } else {
+        spanButton.innerText = 'Admin';
       }
-    }, error => {
-      console.log(error);
-    });
+      li.appendChild(spanButton);
+      list.appendChild(li);
+    }
+  }
+
+  list.addEventListener('click',  (event) => {
```

Comment:
This is called _event delegation_. Rather than attaching an event to each one of the users we attach one single event to the list. We then check that `target` of the event to perform one action or other.